crates/core/component/dex/src/component/rpc.rs

```diff
@@ -193,6 +205,77 @@ impl QueryService for Server {
         }
     }
 
+    #[instrument(skip(self, request))]
+    async fn candlestick_data(
+        &self,
+        request: tonic::Request<CandlestickDataRequest>,
+    ) -> Result<tonic::Response<CandlestickDataResponse>, Status> {
+        let state = self.storage.latest_snapshot();
+        let start_height = request.get_ref().start_height;
+        let limit = request.get_ref().limit as usize;
```

Comment:
We should use a `std::cmp::max` to clamp this to a constant, say 20_000 (= 100k seconds, just longer than 1 day) so that an RPC caller can't DoS the node by forcing it to accumulate giant amounts of data in memory